firebase-firestore/src/androidTest/java/com/google/firebase/firestore/CountTest.java

```diff
@@ -0,0 +1,111 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.firestore;
```

Comment:
Add a test for running a count query when disableNetwork() has disabled the network. I'm not 100% sure what the expected behavior would be here. I would expect the query to fail immediately, but I'm not sure if transactions take enable/disable network into account.